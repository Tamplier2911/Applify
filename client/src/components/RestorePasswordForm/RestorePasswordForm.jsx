import "./RestorePasswordForm.scss";
import React from "react";

const RestorePasswordForm = () => {
  return (
    <div>
      <div>Restore Password Form</div>
    </div>
  );
};

export default RestorePasswordForm;
