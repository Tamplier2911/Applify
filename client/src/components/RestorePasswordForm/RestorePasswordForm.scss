/*
.restore-password-form {
  display: grid;
  grid-row-gap: 4rem;
  grid-auto-rows: min-content;

  &__title {
    // ${getFont}
    font-weight: 400;
    font-size: 1.7rem;

    margin-top: 2rem;
    @media only screen and (max-width: 425px) {
      margin-top: 5rem;
    }

    // remove that when font is getten
    font-family: var(--font-title);
  }

  &__form {
    display: grid;
    grid-row-gap: 2rem;
  }
}
*/
