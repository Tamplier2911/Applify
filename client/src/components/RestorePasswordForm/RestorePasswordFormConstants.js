const restorePasswordFormData = {
  eng: {
    restorePasswordFormTitle: "Please, enter new password.",
    restorePasswordFormNewPass: "New Password",
    restorePasswordFormNewPassConf: "Confirm New Password",
    restorePasswordFormButton: "Restore",
  },
  rus: {
    restorePasswordFormTitle: "Пожалуйста, введите новый пароль.",
    restorePasswordFormNewPass: "Новый пароль",
    restorePasswordFormNewPassConf: "Подтвердите новый пароль",
    restorePasswordFormButton: "Восстановить",
  },
  ukr: {
    restorePasswordFormTitle: "Будь ласка, введіть новий пароль.",
    restorePasswordFormNewPass: "Новий пароль",
    restorePasswordFormNewPassConf: "Підтвердіть новий пароль",
    restorePasswordFormButton: "Відновити",
  },
};

export default restorePasswordFormData;
