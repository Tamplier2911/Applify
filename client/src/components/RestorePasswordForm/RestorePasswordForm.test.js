import React from "react";
import toJson from "enzyme-to-json";

// component
import RestorePasswordForm from "./RestorePasswordForm";

// utils
import {
  setShallowWRC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<RestorePasswordForm />", () => {
  const initialState = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialState);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowWRC(<RestorePasswordForm store={store} />);
  });

  it("renders without an error", () => {
    const restorePasswordForm = findByTestAttr(
      wrapper,
      "restore-password-form"
    );
    expect(restorePasswordForm).toHaveLength(1);
  });

  it("renders children - form element", () => {
    const restorePasswordFormElement = findByTestAttr(
      wrapper,
      "restore-password-form-element"
    );
    expect(restorePasswordFormElement).toHaveLength(1);
  });

  it("renders children - form input", () => {
    const restorePasswordFormInput = findByTestAttr(
      wrapper,
      "restore-password-form-input"
    );
    expect(restorePasswordFormInput).toHaveLength(2);
  });

  it("renders children - form submit", () => {
    const restorePasswordFormSubmit = findByTestAttr(
      wrapper,
      "restore-password-form-submit"
    );
    expect(restorePasswordFormSubmit).toHaveLength(1);
  });

  it("to match snapshot", () => {
    const restorePasswordForm = findByTestAttr(
      wrapper,
      "restore-password-form"
    );
    expect(toJson(restorePasswordForm)).toMatchSnapshot();
  });
});
