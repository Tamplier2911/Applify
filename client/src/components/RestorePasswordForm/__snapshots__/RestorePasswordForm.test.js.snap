// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestorePasswordForm /> to match snapshot 1`] = `
<styled.div
  data-test="restore-password-form"
>
  <styled.h3
    lang="eng"
  >
    Please, enter new password.
  </styled.h3>
  <styled.form
    autoComplete="off"
    data-test="restore-password-form-element"
    onSubmit={[Function]}
  >
    <FormInput
      data-test="restore-password-form-input"
      label="New Password"
      name="newPassword"
      onInputChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <FormInput
      data-test="restore-password-form-input"
      label="Confirm New Password"
      name="newPasswordConfirm"
      onInputChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <Button
      data-test="restore-password-form-submit"
      type="submit"
      value="Restore"
    />
  </styled.form>
</styled.div>
`;
